mansur_adam_ndvi.py

```diff
@@ -0,0 +1,892 @@
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_ts_single_site.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    single_scene_points += 2
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_ts_single_site.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    single_scene_points += 2
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Ensure the site name is correct
+if student_ndvi_ts_single_site.site.values[0] == 'HARV':
+    print('\u2705 You have the correct site name!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct site name.')
+
+if np.allclose(0.281131628228094, student_ndvi_ts_single_site.mean_ndvi.values[0]):
+    print('\u2705 You have the correct mean NDVI value!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct mean ndvi value.')
+
+print("\n \u27A1 You received {} out of 15 points for creating a dataframe.".format(
+    single_scene_points))
+single_scene_points
+
+
+# ## Task 2:
+# 
+# In the cell below, process all of the landsat scenes. Create a DataFrame that contains the following 
+# information for each scene
+# 
+# 
+# |   | index  | site  | mean_ndvi  | 
+# |---|---|---|---|
+# | Date  |   |   |   |
+# |  2017-01-07  | 0  | SJER  | .4  |  
+# 
+# Be sure to call your dataframe at the end of the cell to ensure autograding works.
+# HINT: FOR THIS STEP, leave any rows containing missing values (`NAN`).
+
+# ### Data sources
+# 
+# Data for the following calculation of NDVI was collected by the Landsat 8
+# satellite in 2017 for the HARV and SJER field sites as part of the NEON program.
+
+# In[8]:
+
+
+# Create dataframe of NDVI including the cleaning data to deal with clouds
+
+# Important: to use the ungraded tests below as a sanity check,
+# name your columns: mean_ndvi and site
+# Don't forget to set date as the index and make the values of type datetime
+
+# Define path to SJER folder (the path to HARV is defined above)
+sjer_path = os.path.join("ndvi-automation", "sites", "SJER")
+
+# Read boundaries for each site
+boundaries = {
+    "HARV": gpd.read_file(os.path.join(harv_path, "vector", "HARV-crop.shp")),
+    "SJER": gpd.read_file(os.path.join(sjer_path, "vector", "SJER-crop.shp"))
+}
+
+# Calculate NDVI for each scene
+ndvi_by_scene = []
+for root, dirs, files in os.walk(os.path.join("ndvi-automation", "sites")):
+    if os.path.basename(root).startswith("LC08"):
+        # Grab site name from root and process scene
+        site_name = re.search(r"\b[A-Z]{4}\b", root).group()
+        ndvi_by_scene.append(
+            mean_ndvi_from_landsat(root, site_name, boundaries[site_name])
+        )
```

Comment:
Would this loop be interrupted if there were a directory that didn't start with "LC08"? i.e. do you need to add an "else" in there? I don't actually know the answer to this :)